portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -46,11 +47,34 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         Query query = new Query(DATASTORE_LABEL);
         PreparedQuery results = datastore.prepare(query);
 
-        ArrayList<String> commentList = new ArrayList<String>();
-        for (Entity entity : results.asIterable()) {
-            commentList.add((String) entity.getProperty(INPUT_NAME));
+        int numComments;
+        // Get the number of comments the user wants to view. 
+        try {
+            numComments = Integer.parseInt(request.getParameter(NUM_COMMENTS_INPUT));
+        } catch (NumberFormatException e) {
+            numComments = 20;
         }
 
+        ArrayList<Map<String, String>> commentList = new ArrayList<Map<String, String>>();
+        int counter = 0;
+
+        // Add the users desired amount of non-empty comments.
+        for (Entity entity : results.asIterable()) {
```

Comment:
It might make sense to break this off into its own function? 

Like ArrayList<Comment> loadComments(int numCommentsToLoad)

Reply:
OK